### PR TITLE
[test-only] ApiTest. search by date

### PR DESCRIPTION
**Pull request by ScharfViktor**
- related https://github.com/owncloud/ocis/pull/7263

### CODE REVIEW

**Comment by fschade** [comment]
short note, the natural language date tokens are only supported by the parser at the moment, not sure if everything is already working perfectly in the blevy engine, could need some work!

your tests help a lot, ill take them as cases when checking the implementation!

File: tests/acceptance/features/apiSearch/dateSearch.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      | Mtime:"last month" | /lastMonth.txt  |                 | /today.txt      |                 |
+      | Mtime:"this year"  | /today.txt      |                 | /lastYear.txt   |                 |
+      | Mtime:"last year"  | /lastYear.txt   |                 | /today.txt      |                 |
+      #  500 error. can not compile the query
+      # | Mtime:$today       | /today.txt      |                 | /yesterday.txt  | /lastWeek.txt   |
+      | Mtime>$today       | /today.txt      |                 | /yesterday.txt  |                 |
+      | Mtime>$yesterday   | /today.txt      | /yesterday.txt  |                 |                 |
+      | Mtime>=$yesterday  | /today.txt      | /yesterday.txt  |                 |                 |
+      # 400 bad request. XML syntax error on line 28: expected attribute name in element
+      # | Mtime<$yesterday   | /yesterday.txt  |                 | /today.txt      |                 |
```
[the remark this answers is not in the document]
Comment (2403905) [reply tests/acceptance/features/apiSearch/dateSearch.feature:53]: In this case the `<` has to be escaped 
https://www.advancedinstaller.com/user-guide/xml-escaped-chars.html

**Comment by 2403905** [comment]
@fschade  @ScharfViktor 
When the user opens the file details he can see the `Last modified` date with the minute precision.
Because the bleve can perform date range searches only and doesn't support date equality we can process the kql `Mtime:"2023-09-21T12:45:03Z"` like the bleve query '+Mtime:>="2023-09-21T12:45:03Z" +Mtime:<"2023-09-21T12:45:04Z"'

File: tests/acceptance/features/apiSearch/dateSearch.feature [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      | Mtime:yesterday    | /yesterday.txt  |                 | /today.txt      |                 |
+      | Mtime:"this week"  | /today.txt      |                 | /lastWeek.txt   | /lastMont.txt   |
+      # not implemented
+      # | Mtime:"last week"  | /lastWeek.txt   |                 | /today.txt      |                 |
+      | Mtime:"this month" | /today.txt      |                 | /lastMont.txt   |                 |
+      | Mtime:"last month" | /lastMonth.txt  |                 | /today.txt      |                 |
+      | Mtime:"this year"  | /today.txt      |                 | /lastYear.txt   |                 |
+      | Mtime:"last year"  | /lastYear.txt   |                 | /today.txt      |                 |
+      #  500 error. can not compile the query
+      # | Mtime:$today       | /today.txt      |                 | /yesterday.txt  | /lastWeek.txt   |
```
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/acceptance/features/apiSearch/dateSearch.feature]: https://github.com/owncloud/ocis/pull/7310#issuecomment-1729558051 
I agree, it's a bad case. it's unlikely that user will be search for a file that was created to the exact second. 
I deleted it 

I suggest to delete line:
https://github.com/owncloud/ocis/pull/7263/files#diff-b7932773dddddc5de52acd053d84a1a56cf56907ac142cfe480c3186db942ca8R13
and change line 
https://github.com/owncloud/ocis/pull/7263/files#diff-b7932773dddddc5de52acd053d84a1a56cf56907ac142cfe480c3186db942ca8R16

like `Mtime&lt;"2023-09-05T08:42:11.23554+02:00"`

**Review by 2403905 — APPROVED** [review]

**Pull request merged**